onnxscript/ir/_tape.py

```diff
@@ -2,19 +2,49 @@
 # Licensed under the MIT License.
 """Convenience methods for constructing the IR."""
 
-# NOTE: This is a temporary solution for constructing the IR. It should be replaced
-# with a more permanent solution in the future.
-
 from __future__ import annotations
 
-from typing import Any, Iterable, Iterator, List, Mapping, Optional, Sequence, Tuple
+from typing import (
+    Any,
+    Iterable,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+)
 
 from onnxscript import ir
-from onnxscript.ir import _convenience
-
-
-class Tape(Iterable[ir.Node]):
-    """A tape for recording nodes that are created."""
+from onnxscript.ir import _convenience, _core
+
+
+class Tape(Iterable[_core.Node]):
+    """Tape class.
+
+    A tape is a recorder that collects nodes and initializers that are created so
```

Comment:
It looks like a intermediate container when you don(t know yet if this is going to end up as a graph or a function?

Reply:
Yes. It is just a container so users don't need to keep track of the nodes created.